File: Python For Everyone, Horstmann/Chapter5-Functions/P5.23.py
# Write a program that prints instructions to get coffee, asking the user for input
# whenever a decision needs to be made. Decompose each task into a function, for
# example:
#     def brewCoffee() :
#         print("Add water to the coffee maker.")
#         print("Put a filter in the coffee maker.")
#         grindCoffee()
#         print("Put the coffee in the filter.")
#         . . .

# FUNCTIONS
def grindCoffee():
    choice = str(input("Grind the coffee?(Y/N) "))

    if choice == "y" or choice == "Y":
        print("Add beans to grinder")
        print("Grind up coffee beans for 2 minutes")

def brewCoffee():
    print("Switch the coffee maker on")
    print("Wait for the water to run through")

def pourCoffee():
    print("Get a mug")
    print("Pour coffee in a mug")

def drinkCoffee():
    print("Drink the coffee you just made")

# main
def main():
    print("Add water to the coffee maker")
    print("Put a filter in the coffee maker")
    grindCoffee()
    print("Put coffee in the filter")
    brewCoffee()
    pourCoffee()
    drinkCoffee()


# PROGRAM RUN
main()

File: Big-Java-for-Everyone-Late-Objects-Horstmann/Chapter-5-Methods/P5_23.java
/*Write a program that prints instructions to get coffee, asking the user for input
whenever a decision needs to be made. Decompose each task into a method, for
example:
    public static void brewCoffee()
    {
        System.out.println(“Add water to the coffee maker.”);
        System.out.println(“Put a filter in the coffee maker.”);
        grindCoffee();
        System.out.println(“Put the coffee in the filter.”);
        . . .
    }*/

import java.util.Scanner;

public class P5_23 {
    public static void grindCoffee() {
        System.out.println("Add coffee beans to the coffee grinder.");
        System.out.println("Push the button for 60 seconds.");
    }
    public static void brewCoffee() {
        System.out.println("Add water to the coffee maker.");
        System.out.println("Put a filter in coffee maker.");
        grindCoffee();
        System.out.println("Add coffee beans to filer.");
        System.out.println("Trun coffee maker on.");
    }
    public static void boilWater() {
        System.out.println("Is there a microwave available? (yes/no) ");
        Scanner input = new Scanner(System.in);
        String result = input.next();
        if (result.equals("yes"))
        {
            System.out.println("Fill cup with water.");
            System.out.println("Place cup in microwave.");
            System.out.println("Heat for three minutes.");
        }
        else
        {
            System.out.println("Fill a kettle with water.");
            System.out.println("Heat kettle on stove until water comes " + "to a boil.");
        }
        input.close();
    }
    public static void makeCoffee() {
        System.out.println("Is there instant coffee available? (yes/no)");
        Scanner input = new Scanner(System.in);
        String result = input.next();
        if (result.equals("yes"))
        {
            boilWater();
            System.out.println("Mix boiling water with instant coffee.");
        }
        else
        {
            brewCoffee();
        }
        input.close();
    }
    public static void getCoffee() {
        System.out.println("Can you ask someone? (yes/no) ");
        Scanner input = new Scanner(System.in);
        String result = input.next();
        if (result.equals("no"))
        {
            System.out.println("You must make coffee.");
            makeCoffee();
        }
        else
        {
            System.out.println("Ask for some.");
        }
        input.close();
    }
}
